homeassistant/components/telegram_bot/__init__.py

```diff
@@ -377,34 +393,64 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Telegram bot component."""
     domain_config: list[dict[str, Any]] = config[DOMAIN]
 
-    if not domain_config:
-        return False
-
-    platforms = await async_get_loaded_integration(hass, DOMAIN).async_get_platforms(
-        {p_config[CONF_PLATFORM] for p_config in domain_config}
+    integration: Integration = async_get_loaded_integration(hass, DOMAIN)
+    modules: dict[str, ModuleType] = await integration.async_get_platforms(
+        [PLATFORM_BROADCAST, PLATFORM_POLLING, PLATFORM_WEBHOOKS]
     )
+    MODULES.update(modules)
+
+    # create config entries from YAML
+    for config_import in domain_config:
```

Comment:
Only import the last item, since that is what has been working so far. We don't want to import more since it may break the actions. The user can set up further config entries manually if they want.